### PR TITLE
[RHPAM-3638] - App Fomer maven integration was not honoring the nonPr…

### PR DESCRIPTION
**Pull request by spolti**
[RHPAM-3638] - App Fomer maven integration was not honoring the nonProxyHosts property

Signed-off-by: spolti <fspolti@redhat.com>

### CODE REVIEW

**Comment by ederign** [comment]
@mareknovotny @spolti wants to include this on 7.52.x. Are this branch still open?

File: kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java [hunk 1]
```diff
@@ -155,11 +162,15 @@ private boolean isProfileActive( Profile profile ) {
                                                      .addPassword( server.getPassword() )
                                                      .build() );
         }
-        if (settings.getActiveProxy() != null) {
-            remoteBuilder.setProxy(getActiveAetherProxyFromSettings(settings));
+
+        if (activeProxy != null) {
+            if (null == activeProxy.getNonProxyHosts()) {
+                remoteBuilder.setProxy( getActiveAetherProxyFromSettings(settings));
+            } else if (!repositoryUrlMatchNonProxyHosts(settings.getActiveProxy().getNonProxyHosts(), remoteBuilder.build().getUrl())) {
```
Comment (jesuino) [inline kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java:169]: instead `settings.getActiveProxy()` use the `activeProxy` local variable.

File: kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java [hunk 2]
```diff
@@ -155,11 +162,15 @@ private boolean isProfileActive( Profile profile ) {
                                                      .addPassword( server.getPassword() )
                                                      .build() );
         }
-        if (settings.getActiveProxy() != null) {
-            remoteBuilder.setProxy(getActiveAetherProxyFromSettings(settings));
+
+        if (activeProxy != null) {
+            if (null == activeProxy.getNonProxyHosts()) {
+                remoteBuilder.setProxy( getActiveAetherProxyFromSettings(settings));
```
Comment (jesuino) [inline kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java:168]:
```suggestion
                remoteBuilder.setProxy(getActiveAetherProxyFromSettings(settings));
```
Removing extra space

File: kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java (hunk 1 above)
Comment (jesuino) [reply kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java:169]:
```suggestion
            } else if (!repositoryUrlMatchNonProxyHosts(activeProxy.getNonProxyHosts(), remoteBuilder.build().getUrl())) {
```

File: kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java [hunk 3]
```diff
@@ -155,11 +162,15 @@ private boolean isProfileActive( Profile profile ) {
...
+            } else if (!repositoryUrlMatchNonProxyHosts(settings.getActiveProxy().getNonProxyHosts(), remoteBuilder.build().getUrl())) {
+                remoteBuilder.setProxy( getActiveAetherProxyFromSettings( settings ) );
```
Comment (jesuino) [inline kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java:170]:
```suggestion
                remoteBuilder.setProxy(getActiveAetherProxyFromSettings(settings));
```

Removing extra space

File: kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java [hunk 4]
```diff
@@ -205,12 +217,29 @@ private ArtifactRepository toArtifactRepository( RemoteRepository remoteReposito
             artifactRepository.setAuthentication( new Authentication( server.getUsername(),
                                                                       server.getPassword() ) );
         }
-        if (settings.getActiveProxy() != null) {
-            artifactRepository.setProxy(getActiveMavenProxyFromSettings(settings));
+
+        if (activeProxy != null) {
+            if (null == activeProxy.getNonProxyHosts()) {
+                artifactRepository.setProxy(getActiveMavenProxyFromSettings(settings));
+            } else if (!repositoryUrlMatchNonProxyHosts(settings.getActiveProxy().getNonProxyHosts(), artifactRepository.getUrl())) {
```
Comment (jesuino) [inline kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java:224]:
```suggestion
            } else if (!repositoryUrlMatchNonProxyHosts(activeProxy.getNonProxyHosts(), artifactRepository.getUrl())) {
```

**Review by jesuino — CHANGES_REQUESTED** [review]
Thanks for your great work on this, @spolti  - Here are a few suggestions for improvements

**Review by jesuino — APPROVED** [review]
LGTM!

**Review by sutaakar — APPROVED** [review]
Looks good to me, works as expected.

**Comment by mareknovotny** [comment]
so jira was retargeted to next major release and we can close this

**Pull request closed without merging**